src/js/program/program-controller.js

```diff
@@ -89,6 +89,7 @@ class ProgramController extends Eventable {
                     return nextMediaController;
                 }
             });
+        model.setActiveItem(index);
```

Comment:
This change breaks the test I added over the weekend that ensure that "playlistItem" is triggered before other events.

I think we should remove this change to when `model.setActiveItem(index);`, and instead add an "itemReady" property to the model, that is set to `false` at the beginning of `setActiveItem(index)` and then set to `true` where we trigger "itemReady". This property would be used as the condition for queuing commands on the controller so that `play` when called on playlistItem, does not execute until everything in this method is resolved.

Reply:
@karimJWP @johnBartos 
https://github.com/jwplayer/jwplayer/blob/master/src/js/controller/controller.js#L908-L919

This is the condition under which we wait to execute `play`. It seems like since `activeProvider` is set in between items, play can be run while `setActiveItem` is running. Having an "itemReady" flag would help us prevent this.

```js
        // Setup ApiQueueDecorator after instance methods have been assigned 
        const apiQueue = new ApiQueueDecorator(this, [
            'play',
            'pause',
            'seek',
            'setCurrentAudioTrack',
            'setCurrentCaptions',
            'setCurrentQuality',
            'setFullscreen',
        ], () => !_programController.activeProvider);
```